src/editor/Db.js

```diff
@@ -0,0 +1,386 @@
+  *                                                                   *
+  *     The db module interacts with the editor while shadowing its   *
+  *		assigned codemirror for a given doc in order to preserve any  *
+  *		unsaved changes. This module features database config	 	  *
+  *		info, and methods for db instantiation and CRUD. There are	  *
+  *		four tables created in total amounting to 500MB in total. 	  *
+  *		These tables are each respectively named "cursorpos_coords",  *
+  *		"scrollpos_coords", "undo_redo_history" and 				  *
+  *		"unsaved_doc_changes". In order, these contain the last	 	  *
+  *		known document information as related to cursor	  			  *
+  *		and scroll positioning, as well as the undo/redo history 	  *
+  *		and document text. Change data is stored by full filepath 	  *
+  *		which is used as the sessionId. Keyup events in a focused 	  *
+  *		editor are what trigger syncing of all current history 		  *
+  *		data to the database. 									      *
+  *                                                                   *
+  \*******************************************************************/
+define(function (require, exports, module) {
+    'use strict';
+
+    var Editor = require("editor/Editor"),
+        PreferencesManager = require("preferences/PreferencesManager"),
+    	Strings = require("strings"),
+        DocumentManager = require("document/DocumentManager"),
+    	CompressionUtils = require("thirdparty/rawdeflate"),
+        CompressionUtils = require("thirdparty/rawinflate"),
+        He = require("thirdparty/he");
+
+    // Config settings
+    var DB_NAME    = 'change_history_db',
+        DB_VERSION = '1.0',
+        DB_DESC    = 'Feature: Hot Close',
+        DB_SIZE_MB = 500;
+
+    var database   = window.openDatabase(DB_NAME,
+                                    DB_VERSION,
+                                    DB_DESC,
+                                    DB_SIZE_MB * 1024 * 1024);
+
+    // Static db references
+    var tables = [
+            "cursorpos_coords",
+            "scrollpos_coords",
+            "undo_redo_history",
+            "unsaved_doc_changes"
+        ],
+        keyNames = [
+            "int__CursorPos",
+            "int__ScrollPos",
+            "str__DocHistory",
+            "str__DocTxt"
+        ];
+
+    // Initialize 'hot close' setting
+    var HOT_CLOSE = "hotClose";
+
+    PreferencesManager.definePreference(HOT_CLOSE, "boolean", true, {
+        description: Strings.DESCRIPTION_HOT_CLOSE
+    });
+
+    var hotClose = PreferencesManager.get(HOT_CLOSE);
+    
+    // Debounce syncing of new unsaved changes to db
+    var timer = null;
+    function debouncedSync(doc, delay) {
+		var result = new $.Deferred();
+		try {
+
+			return function () {
+				clearTimeout(timer);
+				timer = setTimeout(function () {
+					captureUnsavedDocChanges(doc);
+				}, delay || 1250);
+				result.resolve();
+			};
+		} catch (error) {
+			console.log(error);
```

Comment:
I do not believe that it is a good idea just to send the error to the console. If everyone does that, the console will be a bit messy. These errors should be properly threated in the application. As I do not have experience with the platform, I will keep the point to @nethip, but this is just a comment that can help you to think while waiting for a proper review. Note that the same happens in some other areas of your contribution.